### PR TITLE
Add operation definitions

### PR DESCRIPTION
**Pull request by patrickkwang**

### CODE REVIEW

**Review by edeutsch — APPROVED** [review]

**Review by dkoslicki — CHANGES_REQUESTED** [review]
Patrick said he will investigate a machine readable YAML and YAML-to-markdown autogen documentation

File: docs/build/op_template.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+{% endif -%}{{ data.description }}
+
+{% if data.examples -%}
+### examples
+
+{% for example in data.examples -%}
+- [input](../examples/{{ example.input }}), [output](../examples/{{ example.output }})
+{% endfor %}
+{% endif %}### input requirements
```
Comment (dkoslicki) [inline docs/build/op_template.md:14]: Was a newline intended here?

File: docs/build/index_template.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Operation Index
+
+{% for op in operations -%}
```
Comment (dkoslicki) [inline docs/build/index_template.md:3]: Not quite clear to me what these `for` loops and things like that are here for. Perhaps we can clarify in the call today. Perhaps it's Jinja magic?

**Review by dkoslicki — APPROVED** [review]

File: docs/build/op_template.md (hunk 1 above)
Comment (patrickkwang) [reply docs/build/op_template.md:14]: I find whitespace management in jinja really annoying. It looks weird here, but I think this produces the output that I want.

**Pull request merged**